sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+ *
+ * {@codesnippet com.azure.storage.blob.BlobBatch.createWithServiceClient}
+ * {@codesnippet com.azure.storage.blob.BlobBatch.createWithServiceAsyncClient}
+ *
+ * <p>Azure Storage Blob batches are homogeneous which means a {@link #delete(String) delete} and {@link
+ * #setTier(String, AccessTier) set tier} are not allowed to be in the same batch.</p>
+ *
+ * {@codesnippet com.azure.storage.blob.BlobBatch.illegalBatchOperation}
+ *
+ * <p>Please refer to the <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch">Azure Docs</a>
+ * for more information.</p>
+ */
+public final class BlobBatch {
+    private static final String BATCH_REQUEST_CONTENT_ID = "Batch-Request-Content-Id";
+    private static final String BATCH_REQUEST_URL_PATH = "Batch-Request-Url-Path";
+    private static final String CONTENT_ID = "Content-Id";
+    private static final String BATCH_BOUNDARY_TEMPLATE = "batch_%s";
+    private static final String CONTENT_TYPE = "Content-Type: application/http";
+    private static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding: binary";
+    private static final String CONTENT_ID_TEMPLATE = "Content-ID: %d";
+    private static final String HTTP_VERSION = "HTTP/1.1";
+    private static final String OPERATION_TEMPLATE = "%s %s %s";
+    private static final String HEADER_TEMPLATE = "%s: %s";
+    private static final String NEWLINE = "\r\n";
+
+    private static final int[] EXPECTED_DELETE_STATUS_CODES = { 202 };
+    private static final int[] EXPECTED_SET_TIER_STATUS_CODES = { 200, 202 };
+
+    private final ClientLogger logger = new ClientLogger(BlobBatch.class);
+
+    private final BlobAsyncClient batchClient;
+
+    private final Deque<Mono<? extends Response<?>>> batchOperationQueue;
+    private final List<ByteBuffer> batchRequest;
+    private final Map<Integer, BlobBatchOperationResponse<?>> batchMapping;
+
+    private final AtomicInteger contentId;
+    private final String batchBoundary;
+
+    private BlobBatchType batchType;
+
+    /**
+     * Constructs a {@link BlobBatch} using the {@link BlobServiceClient#getHttpPipeline()
+     * BlobServiceClient's HttpPipeline} to build a modified {@link HttpPipeline} that is used to prepare the requests
+     * in the batch.
+     *
+     * @param client {@link BlobServiceClient} used to construct the batch.
+     */
+    public BlobBatch(BlobServiceClient client) {
+        this(client.getAccountUrl(), client.getHttpPipeline());
+    }
+
+    /**
+     * Constructs a {@link BlobBatch} using the {@link BlobServiceAsyncClient#getHttpPipeline()
+     * BlobServiceAsyncClient's HttpPipeline} to build a modified {@link HttpPipeline} that is used to prepare the
+     * requests in the batch.
+     *
+     * @param client {@link BlobServiceAsyncClient} used to construct the batch.
+     */
+    public BlobBatch(BlobServiceAsyncClient client) {
```

Comment:
I am not sure how users will interact with this but from looking at this, there's a batch that can be created using sync or async client but all the APIs here are returning sync types.

Reply:
Removed constructors for this class, need to get the `BlobBatch` from the client.